datafusion/sqllogictest/test_files/expr.slt

```diff
@@ -2288,39 +2288,39 @@ select struct(time,load1,load2,host) from t1;
 
 # can have an aggregate function with an inner coalesce
 query TR
-select t2.info['c3'] as host, sum(coalesce(t2.info)['c1']) from (select struct(time,load1,load2,host) as info from t1) t2 where t2.info['c3'] IS NOT NULL group by t2.info['c3'] order by host;
+select t2."struct(t1.time,t1.load1,t1.load2,t1.host)"['c3'] as host, sum(coalesce(t2."struct(t1.time,t1.load1,t1.load2,t1.host)")['c1']) from (select struct(time,load1,load2,host) from t1) t2 where t2."struct(t1.time,t1.load1,t1.load2,t1.host)"['c3'] IS NOT NULL group by t2."struct(t1.time,t1.load1,t1.load2,t1.host)"['c3'] order by host;
```

Comment:
This reverts the change from https://github.com/apache/arrow-datafusion/pull/9894